caps/capabilities.go

```diff
@@ -222,3 +242,17 @@ func (e *NotFoundError) Error() string {
 		panic(fmt.Sprintf("unexpected what: %q", e.what))
 	}
 }
+
+// Validate if a capability is correct according to the given type
+func (t Type) Validate(c *Capability) error {
+	if t != c.Type {
+		return fmt.Errorf("capability is not of type %q", t)
+	}
+	// While we don't have any support for type-specific attribute schema,
+	// let's ensure that attributes are totally empty. This will make tests
+	// show that this code is actually being used
+	if c.Attrs != nil && len(c.Attrs) != 0 {
```

Comment:
FYI, no need to check whether c.Attrs is not nil before testing its length. A nil map has len zero. Also, > 0 avoids the negation, which means it's easier for humans to read, but now I'm bikesheding. :-)